### ISSUE TITLE
IOException: Error 183 when processing asset packs

### ISSUE THREAD

**Issue by tutipeti** [google/play-unity-plugins#69]
Hey,

our Unity build on CI just failed with the following output:

```
Build succeeded
BuildPlayer "succeeded" with 1 error(s). Assuming success. 

Processing asset pack 1 of 14...
IOException: Error 183 
(Filename: <fb001e01371b4adca20013e0ac763896> Line: 0)
```

The [MSDN docs](https://docs.microsoft.com/en-us/windows/win32/debug/system-error-codes--0-499-) say the following about this error:
```
ERROR_ALREADY_EXISTS

183 (0xB7)

Cannot create a file when that file already exists.
```

My guess is that the temp path returned by `GetTempFolder` in [AppBundlePublisher](https://github.com/google/play-unity-plugins/blob/master/GooglePlayPlugins/com.google.android.appbundle/Editor/Scripts/Internal/AppBundlePublisher.cs#L216) might clash, and we might have had 2 builds running at the same time on the same machine.

I tried calling that method now on my machine several times and it always returns the same path in OSX.

Any chance you could change that by adding some uniqueness to the path?

**Comment by Neu2o** [comment]
Well, `GetTempFolder` simply calls `Path.GetTempPath` and this never returns "a new temp directory", but "_the_ temp folder" of the user; which is the same for every call. See [MSDN docs](https://docs.microsoft.com/en-us/dotnet/api/system.io.path.gettemppath).

If you want a new temp directory, you need something like `Path.Combine(Path.GetTempPath(), Path.GetRandomFileName())`.

There are some discussions out there wether this can possibly collide with random names by other processes, but asking me this is negligible in our use-case, especially when combined with the `play-unity-build` folder. In cases it's not you could add additional checks wether such a directory already exist and retry or you use `GetTempFileName`, take the basename of it, and remove the created random file. But as said, I feel the first approach is good enough.

BTW if you change `GetTempFolder` to return a new random directory, you might want to change [this here](https://github.com/google/play-unity-plugins/blob/f5f4261377363d0c22ef6519506bfad98e50bc85/GooglePlayPlugins/com.google.android.appbundle/Editor/Scripts/Internal/AppBundlePublisher.cs#L112) to `appBundleBuilder.WorkingDirectoryPath` to create `temp.aab` in the same working dir.

**Comment by thanwah** [comment]
Thanks @tutipeti and @Neu2o for creating and commenting on this issue.

This will be fixed in the next release (1.4.0) expected to ship next week (by March 12th).

**Comment by thanwah** [comment]
This was fixed [in v1.4.0](https://github.com/google/play-unity-plugins/releases/tag/v1.4.0), but please let me know if you run into any issues.
